genet/schedule_elements.py

```diff
@@ -287,22 +372,74 @@ def __str__(self):
 
     def _build_graph(self, stops: List[Stop]):
         route_graph = nx.DiGraph(name='Route graph')
-        route_nodes = [(stop.id, stop.__dict__) for stop in stops]
+        try:
+            route_nodes = [(stop.id, stop.__dict__) for stop in stops]
+            stop_edges = [(from_stop.id, to_stop.id) for from_stop, to_stop in zip(stops[:-1], stops[1:])]
+        except AttributeError:
+            route_nodes = [(stop, {}) for stop in stops]
+            stop_edges = [(from_stop, to_stop) for from_stop, to_stop in zip(stops[:-1], stops[1:])]
         route_graph.add_nodes_from(route_nodes, routes=[self.id])
-        stop_edges = [(from_stop.id, to_stop.id) for from_stop, to_stop in zip(stops[:-1], stops[1:])]
         route_graph.add_edges_from(stop_edges, routes=[self.id], modes=[self.mode])
+        route_graph.graph['routes'] = {self.id: self._surrender_to_graph()}
+        route_graph.graph['services'] = {}
+        route_graph.graph['change_log'] = change_log.ChangeLog()
         return route_graph
 
+    def add_additional_attributes(self, attribs: dict):
```

Comment:
necessary as users can now add any attributes they like to routes, services and stops

Reply:
Excellent, some human readable attributes for post processing will be very handy